src/Workspaces/Core/Portable/FindSymbols/Shared/AbstractSyntaxIndex_Persistence.cs

```diff
@@ -39,7 +39,7 @@ internal partial class AbstractSyntaxIndex<TIndex>
     {
         var storageService = project.LanguageServices.SolutionServices.GetPersistentStorageService();
         var documentKey = DocumentKey.ToDocumentKey(ProjectKey.ToProjectKey(solutionKey, project), document);
-        var stringTable = SyntaxTreeIndex.GetStringTable(project);
+        var stringTable = new Lazy<StringTable>(() => SyntaxTreeIndex.GetStringTable(project));
```

Comment:
>Lazy [](http://example.com/codeflow?start=30&length=4)

Bit of a shame there isn't a Lazy that takes in the extra data to avoid the closure